src/docfx/build/xref/XrefResolver.cs

```diff
@@ -15,10 +15,17 @@ internal class XrefResolver
     private readonly DependencyMapBuilder _dependencyMapBuilder;
     private readonly FileLinkMapBuilder _fileLinkMapBuilder;
     private readonly Repository? _repository;
+
+    // the hostname used in output xrefmap
     private readonly string _xrefHostName;
+
     private readonly InternalXrefMapBuilder _internalXrefMapBuilder;
     private readonly Func<JsonSchemaTransformer> _jsonSchemaTransformer;
 
+    // XrefResolver is able to handle both new hostname and legacy hostname in xref urls.
+    private readonly IReadOnlyDictionary<string, string> _hostNameMap; // key: legacy hostname -> value: new hostname
```

Comment:
How about using terms `hostName` and `alternativeHostName`? It looks weird that we will product xref map with `legacyHostName`

Reply:
sure